### PR TITLE
ENH rattler

### PR DESCRIPTION
**Pull request by beckermr**
To do:

- [x] understand and fix timeouts (xref: https://github.com/mamba-org/rattler/pull/711)
- [x] figure out how to add `run_constrains` to the solver calls (xref: https://github.com/mamba-org/rattler/issues/712)

closes #19

### CODE REVIEW

**Comment by wolfv** [comment]
This is cool. Please let us know if you find anything that's wrong with rattler :) 🚀

File: conda_forge_feedstock_check_solvable/rattler_solver.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+                    solution,
+                    _specs,
+                    [MatchSpec(igrf) for igrf in ignore_run_exports_from],
+                    [MatchSpec(igr) for igr in ignore_run_exports],
+                )
+
+        except Exception as e:
+            err = str(e)
+            print_warning(
+                "MAMBA failed to solve specs \n\n%s\n\nwith "
```
Comment (jaimergp) [inline conda_forge_feedstock_check_solvable/rattler_solver.py:134]:
```suggestion
                "RATTLER failed to solve specs \n\n%s\n\nwith "
```

**Comment by beckermr** [comment]
@wolfv @jaimergp This PR is ready to merge. A few notes. 

 - I punted on the `.solv` files with mamba since rattler appears to be working with the test suite here. That is pretty non-trivial actually. The test suite here covers some very complicated feedstocks (arrow-cpp) and depends on properly interpreting `run_constrains` and virtual packages. 
 - I did not use the built-in rattler virtual package functionality and am instead writing them as a channel on disk. This means we can use the same code with mamba. 
 - The timneouts appear to be working with rattler.
 - rattler appears to be caching repodata very well

**Comment by wolfv** [comment]
Thank you @beckermr for pushing this through. This is an awesome test for `resolvo` and it's good to know that it works!

Also happy that we can make use of the features that we added to resolvo especially for these cases (timeout etc.) :)

File: conda_forge_feedstock_check_solvable/mamba_solver.py [hunk 2]
```diff
@@ -41,6 +44,39 @@
...
+def _get_solver(channels, platform, constraints):
+    pool, repos = _get_pool(channels, platform)
+
+    solver_options = [(api.SOLVER_FLAG_ALLOW_DOWNGRADE, 1)]
+    solver = api.Solver(pool, solver_options)
+
+    if constraints:
+        for repo in repos:
+            # need set_installed for add_pin, not sure why
+            repo.set_installed()
```
[the remark this answers is not in the document]
Comment (beckermr) [reply conda_forge_feedstock_check_solvable/mamba_solver.py:72]: To clarify, `SOME_EMPTY_JSON` is a path on disk?

[the remark this answers is not in the document]
Comment (beckermr) [reply conda_forge_feedstock_check_solvable/mamba_solver.py]: Thank you! fixed!

**Pull request merged**